src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -810,22 +815,24 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
                 PathList pathList;
                 lock (m_domainTable)
```

Comment:
Same issue on line 866 below the comment `// (This is the only place that does domain removal)`